### PR TITLE
Enable sonar in junit5 ext repo

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

Enable sonar integration (untested).  Uses same approach as main project.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -37,14 +37,50 @@ jobs:
         with:
           java-version: 17
           distribution: 'zulu'
-          cache: 'maven'
```
Comment (SamBarker) [inline .github/workflows/maven.yml:40]: I don't think we need the fully customised options of `Cache Maven packages` but could be wrong. I think this was fine.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -37,14 +37,50 @@ jobs:
...
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```
Comment (SamBarker) [inline .github/workflows/maven.yml:53]: as mentioned above I don't think we need this block.

File: .github/workflows/sonar.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'zulu'
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```
Comment (SamBarker) [inline .github/workflows/sonar.yaml:63]:
```suggestion
          distribution: 'zulu'
          cache: maven
```

File: .github/workflows/sonar.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      - name: SonarCloud Scan on PR
```
Comment (SamBarker) [inline .github/workflows/sonar.yaml:64]: Ohh thats an "interesting" point, we don't restore the sonar cache

**Review by SamBarker — APPROVED** [review]
LGTM

Though highlights a couple of wrinkles in the runtime repo flow as well.

File: .github/workflows/sonar.yaml (hunk 4 above)
Comment (k-wall) [reply .github/workflows/sonar.yaml:74]: I made a change to address that.

**Comment by SamBarker** [comment]
LGTM

Thanks @k-wall

**Pull request merged**